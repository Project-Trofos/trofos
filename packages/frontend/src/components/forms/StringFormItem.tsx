import React from 'react';
import { Form, Input } from 'antd';
import { LabelTooltipType } from 'antd/lib/form/FormItemLabel';

type FormItemProps = {
  label: string;
  name: string;
  tooltip?: LabelTooltipType;
  isTextArea?: boolean;
  isDisabled?: boolean;
  isRequired?: boolean;
  initialValue?: string;
};

// Input item for any string input
export default function StringFormItem(props: FormItemProps): JSX.Element {
  const { isDisabled, isRequired, label, name, tooltip, initialValue, isTextArea: useTextArea } = props;
  return (
    <Form.Item
      label={label}
      name={name}
      rules={isRequired ? [{ required: true, message: `Please input ${label.toLowerCase()}!` }] : []}
      tooltip={tooltip}
      initialValue={initialValue}
    >
      {useTextArea ? <Input.TextArea rows={4} disabled={isDisabled} /> : <Input disabled={isDisabled} />}
    </Form.Item>
  );
}
