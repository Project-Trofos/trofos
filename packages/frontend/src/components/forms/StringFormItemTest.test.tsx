import React from 'react';
import { render, screen } from '@testing-library/react';
import FormWrapper from './__testutils__/FormRenderHelper.test';
import StringFormItem from './StringFormItem';

describe('SelectCourseCodeFormItem test', () => {
  const LABEL = 'LABEL';
  const NAME = 'NAME';

  function renderForm() {
    return render(
      <FormWrapper>
        <StringFormItem label={LABEL} name={NAME} />
      </FormWrapper>,
    );
  }

  describe('when rendering', () => {
    it('should have correct fields', () => {
      const { baseElement } = renderForm();
      expect(baseElement).toMatchSnapshot();
      const select = screen.getByLabelText(LABEL);
      expect(select).toBeInTheDocument();
    });
  });
});
