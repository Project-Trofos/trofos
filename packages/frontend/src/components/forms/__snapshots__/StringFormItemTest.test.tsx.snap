// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectCourseCodeFormItem test > when rendering > should have correct fields 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class=""
              for="NAME"
              title="LABEL"
            >
              LABEL
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input"
                  id="NAME"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
